tests/test_subscription.py

```diff
@@ -127,9 +166,44 @@ def test_create_and_get_subscription(self, client):
             "/subscription/application",
         )
         assert response.status_code == 200
-        assert response.json().get("credit") == 123
+        assert response.json().get("credit") == 100
         assert response.json().get("active") is True
 
+        # case 2: create subscription with invalid credit
+        new_subscription = SubscriptionIn(
+            username="tester",
+            application="application2",
+            tier=SubscriptionTier.TRIAL,
+            credit=400,
+            expires_at=None,
+            recurring=False,
+        )
+        response = client.post(
+            "/subscription",
+            data=new_subscription.json(),
+        )
+        assert response.status_code == 200
+        response = client.get(
+            "/subscription/application2",
+        )
+        assert response.status_code == 200
+        assert response.json().get("credit") == 200
+
+        # case 3: create subscription with invalid application
+        new_subscription = SubscriptionIn(
+            username="tester",
+            application="not_application",
+            tier=SubscriptionTier.TRIAL,
+            credit=100,
+            expires_at=None,
+            recurring=False,
+        )
+        response = client.post(
+            "/subscription",
+            data=new_subscription.json(),
+        )
+        assert response.status_code == 400
```

Comment:
we can even consider 404 - Application Not Found

Reply:
I think so, let me update this.

Reply:
pushed this change